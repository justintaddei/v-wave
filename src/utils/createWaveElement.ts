import type { IVWaveDirectiveOptions } from '../options'

export const createWaveElement = (x: number, y: number, size: number, options: IVWaveDirectiveOptions) => {
  const waveElement = document.createElement('div')

  waveElement.style.position = 'absolute'
  waveElement.style.width = `${size}px`
  waveElement.style.height = `${size}px`
  waveElement.style.top = `${y}px`
  waveElement.style.left = `${x}px`
  waveElement.style.background = options.color
  waveElement.style.borderRadius = '50%'
  waveElement.style.opacity = `${options.initialOpacity}`
  waveElement.style.transform = 'translate(-50%,-50%) scale(0)'
  waveElement.style.transition = `transform ${options.duration}s ${options.easing}, opacity ${options.duration}s ${options.easing}`

  return waveElement
}
