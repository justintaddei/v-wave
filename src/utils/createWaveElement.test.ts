import { expect, test } from 'vitest'
import type { IVWaveDirectiveOptions } from '../options'
import { createWaveElement } from './createWaveElement'

test('createWaveElement returns a <div>', () => {
  expect(createWaveElement(0, 0, 0, {} as IVWaveDirectiveOptions)).toMatchSnapshot()
})
