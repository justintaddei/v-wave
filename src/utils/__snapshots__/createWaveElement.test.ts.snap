// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createWaveElement returns a <div> 1`] = `
<div
  style="position: absolute; width: 0px; height: 0px; top: 0px; left: 0px; border-radius: 50%; transform: translate(-50%,-50%) scale(0); transition: transform undefineds undefined, opacity undefineds undefined;"
/>
`;
